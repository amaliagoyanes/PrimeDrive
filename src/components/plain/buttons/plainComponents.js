import React from 'react';
import {
    TouchableOpacity,
    Text,
    View,
    TextInput,
    Image,
} from 'react-native';
import {connect} from 'react-redux';
import {injectIntl, intlShape, FormattedMessage} from 'react-intl';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {Col, Row} from 'react-native-easy-grid';
import styles from '../../../style/styles';
import {scale, scaleText} from '../../../scaling';

export class IconButton extends React.Component {
    onPress() {
        if (this.props.measures)
            this._btn.measure((fx, fy, width, height, px, py) => {
                this.props.onPress(px, py, width, height, fx, fy);
            });
        else
            this.props.onPress();
    }

    render() {
        let props = this.props;
        let text = this.props.price;
        if (this.props.intl !== undefined && this.props.intl.messages !== undefined && this.props.intl.messages[this.props.price])
            text = this.props.intl.messages[this.props.price];
        return (
            <Col>
                <TouchableOpacity
                    onLongPress={() => {
                        console.log('this.props.onLongPress');
                        if (this.props.onLongPress)
                            this.props.onLongPress();
                    }} onPress={() => this.onPress()} ref={elt => this._btn = elt}
                    style={props.active ? [styles.button, styles.activeColor, props.style] : [styles.button, props.style]}>
                    {props.image !== undefined &&
                    <Image source={{uri: props.image}} style={[styles.imageOnButton]}></Image>
                    }
                    {props.badge !== undefined &&
                    <View style={styles.iconButtonBadge}>
                        <Text style={[styles.text, {fontSize: scaleText(10)}]}>{props.badge}</Text>
                    </View>
                    }
                    {props.status !== undefined &&
                    <View style={[styles.iconButtonBadge, {backgroundColor: props.status ? '#2dab61' : '#707076'}]}/>
                    }
                    {props.icon !== undefined &&
                    <Icon size={scaleText(18)} name={props.icon} color="white"/>}
                    {props.text !== undefined &&
                    <Text
                        style={[styles.text, {
                            textAlign: 'center',
                            fontSize: scaleText(12)
                        }]}>{props.text.toUpperCase()}</Text>}
                    {props.price !== undefined &&
                    <Text style={[styles.text, {
                        textAlign: 'center',
                        fontSize: scaleText(10)
                    }]}>{text.toUpperCase()}</Text>}
                </TouchableOpacity>
            </Col>
        )
    }
}

export class ButtonItem extends React.Component {
    render() {
        let props = this.props;
        let text = this.props.text;
        if (this.props.intl !== undefined && this.props.intl.messages !== undefined && this.props.intl.messages[this.props.text])
            text = this.props.intl.messages[this.props.text];
        return (
            <Col>
                <TouchableOpacity onPress={props.onPress}
                                  activeOpacity={props.disabled !== undefined && props.disabled ? 1 : 0.2}
                                  style={props.active ? [styles.button, styles.activeColor, props.style] : [styles.button, props.style]}>
                    <Text style={[styles.text, props.textStyle]}>{text.toUpperCase()}</Text>
                    {props.detailsText &&
                    <Text style={[styles.text, props.textStyle, {textAlign: 'right'}]}>{props.detailsText}</Text>
                    }
                    {props.modalActive &&
                    <View
                        style={[styles.modalActiveArrow, {backgroundColor: props.style[props.style.length - 1] ? props.style[props.style.length - 1].backgroundColor : props.active ? '#6f7076' : '#53535a'}]}/>
                    }
                </TouchableOpacity>
            </Col>
        )
    }
}

export class TwoTextButtonItem extends React.Component {
    render() {
        let props = this.props;
        let text = this.props.text;
        if (this.props.intl !== undefined && this.props.intl.messages !== undefined && this.props.intl.messages[this.props.text])
            text = this.props.intl.messages[this.props.text];
        return (
            <Col>
                <TouchableOpacity onPress={props.onPress}
                                  activeOpacity={props.disabled !== undefined && props.disabled ? 1 : 0.2}
                                  style={props.active ? [styles.button, styles.activeColor, props.style] : [styles.button, props.style]}>
                    <Row>
                        <Col style={{width: '70%', marginLeft: scale(10), justifyContent: 'center',}}>
                            <Text
                                style={[styles.text, props.textStyle, {textAlign: 'left'}]}>{text.toUpperCase()}</Text>
                        </Col>
                        {props.detailsText &&
                        <Col style={{marginRight: scale(10), justifyContent: 'center',}}>
                            <Text
                                style={[styles.text, props.textStyle, {textAlign: 'right'}]}>{props.detailsText}</Text>
                        </Col>
                        }
                    </Row>
                </TouchableOpacity>
            </Col>
        )
    }
}


export class ButtonMeasures extends React.Component {
    onPress() {
        this._btn.measure((fx, fy, width, height, px, py) => {
            this.props.onLongPress(px, py);
        });
    }

    render() {
        let props = this.props;
        let text = this.props.text;
        if (this.props.intl !== undefined && this.props.intl.messages !== undefined && this.props.intl.messages[this.props.text])
            text = this.props.intl.messages[this.props.text];
        return (
            <Col>
                <TouchableOpacity onLongPress={() => this.onPress()} ref={elt => this._btn = elt}
                                  onPressIn={() => {
                                      this.lastTimePressIn = new Date().getTime();
                                  }}
                                  onPressOut={() => {
                                      let time = new Date().getTime();
                                      let diff = time - this.lastTimePressIn;
                                      if (diff < 500) {
                                          if (this.props.onPress !== undefined && this.props.onDoublePress === undefined)
                                              this.props.onPress();
                                          else if (this.lastTimePress && time - this.lastTimePress < 1500) {
                                              if (this.props.onDoublePress !== undefined)
                                                  this.props.onDoublePress();
                                              delete this.lastTimePress;
                                          } else {
                                              this.lastTimePress = time;
                                          }
                                      }
                                  }}
                                  style={props.active ? [styles.button, styles.activeColor, props.style] : [styles.button, props.style]}>
                    <Text style={[styles.text, props.textStyle]}>{text.toUpperCase()}</Text>
                    {(props.detailsText &&
                        <Text style={[styles.text, props.textStyle, {textAlign: 'right'}]}>{props.detailsText}</Text>
                    )}
                </TouchableOpacity>
            </Col>
        );
    }
}

export const TableItem = (props) => (
    <Col>
        <TouchableOpacity onPress={props.onPress}
                          style={[styles.button, styles.tableItem, props.style]}>
            <Row style={{alignItems: 'center', justifyContent: 'center'}}>
                <Text style={[styles.text, styles.tableItemNum, props.textStyle]}>{props.num}</Text>
            </Row>

            <Row style={{alignItems: 'center', justifyContent: 'center'}}>
                {(props.time &&
                    <Text
                        style={[styles.text, {
                            fontSize: scaleText(10),
                            color: '#fefefe'
                        }, props.textStyle]}>{props.time}</Text>
                )}
            </Row>
        </TouchableOpacity>
    </Col>
);
