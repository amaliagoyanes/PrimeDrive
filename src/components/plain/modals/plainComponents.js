import moment from 'moment';
import React from 'react';
import {
    TouchableOpacity,
    Text,
    View,
    TextInput
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {Col, Row} from 'react-native-easy-grid';
import styles from '../../../style/styles';
import {scale, scaleText} from '../../../scaling';
import {Timer} from '../index.js';


export const ModalCloseButton = (props) => (
    <Row size={1}>
        <Col size={20}/>
        <Col size={1}>
            <TouchableOpacity onPress={props.close}
                              style={[styles.bigTimesIcon, {
                                  marginTop: 0, marginLeft: scale(32), padding: scale(5),
                                  backgroundColor: 'transparent'
                              }]}>

                <Icon size={scaleText(32)} name="close" color="#313138"/>
            </TouchableOpacity>
        </Col>
    </Row>
);

export class ModalHeader extends React.Component {
    constructor(props) {
        super(props);

        this.getToday = this.getToday.bind(this);
    }


    getToday() {
        return moment().format('DD.MM.YYYY');
    }

    render() {
        let props = this.props;
        return (
            <Row size={1} style={styles.modalHeader}>
                <Col size={1}>
                    <Timer style={{
                        fontSize: scaleText(12), color: '#313138', textAlign: 'left'
                    }}/>
                </Col>
                <Col size={1}>
                    <Text style={[styles.text, {
                        fontSize: scaleText(14), color: '#313138', textAlign: 'center'
                    }]}>{props.intl !== undefined && props.intl.messages[props.modalName]} </Text>
                </Col>
                <Col size={1}>
                    <Text style={[styles.text, {
                        fontSize: scaleText(12),
                        color: '#313138',
                        textAlign: 'right'
                    }]}>{this.getToday()}</Text>
                </Col>
            </Row>
        )
    }
}
