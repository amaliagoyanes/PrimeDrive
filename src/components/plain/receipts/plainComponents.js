import moment from 'moment';
import React from 'react';
import {
    TouchableOpacity,
    Text,
    View,
    TextInput
} from 'react-native';
import {FormattedMessage} from 'react-intl';
import {Col, Row} from 'react-native-easy-grid';
import styles from '../../../style/styles';
import {scale, scaleText} from '../../../scaling';

export const SectionHeader = (props) => (
    <View style={styles.sectionHeader}>
        <Text style={[styles.text, styles.sectionProduct]}>
            {props.intl !== undefined && props.intl.messages["app.product"].toUpperCase()}</Text>
        <Text style={[styles.text, styles.sectionUnits]}>
            {props.intl !== undefined && props.intl.messages["app.units"].toUpperCase()}</Text>
        <Text style={[styles.text, styles.sectionTotalPrice]}>
            {props.intl !== undefined && props.intl.messages["app.totalPrice"].toUpperCase()}</Text>
    </View>
);

export const ReceiptTotals = (props) => (
    <View style={[styles.receiptTotals, props.style]}>
        {props.discount !== undefined && props.discount > 0 &&
        <View style={[styles.receiptTotalsRow, {marginBottom: props.small ? scale(5) : scale(10)}]}>
            <Text
                style={[styles.text, styles.receiptTotalsLeftCol,
                    {fontSize: props.small ? scaleText(10) : scaleText(12)}]}>
                {props.intl !== undefined && props.intl.messages["app.discount"].toUpperCase()}
            </Text>
            <Text style={[styles.text, styles.receiptTotalsRightCol,
                {fontSize: props.small ? scaleText(10) : scaleText(12)}]}>
                -{props.discount}% / -{toLocale((props.total * props.discount / 100).toFixed(2)).replace('.', ',')}
                DKK</Text>
        </View>
        }
        <View style={styles.receiptTotalsRow}>
            <Text style={[styles.text, styles.receiptTotalsLeftCol, {
                fontFamily: 'Gotham-Bold',
                fontSize: props.small ? scaleText(12) : scaleText(16)
            }]}>
                {props.intl !== undefined && props.intl.messages["app.total"].toUpperCase()}</Text>
            <Text style={[styles.text, styles.receiptTotalsRightCol, {
                fontFamily: 'Gotham-Bold',
                fontSize: props.small ? scaleText(12) : scaleText(16)
            }]}>{props.discount !== undefined ? toFixedLocaleDa(props.total * (1 - props.discount / 100)) :
                toFixedLocaleDa(props.total)
            } DKK</Text>
        </View>
    </View>
);

export const ReceiptListViewFooter = (props) => (
    <View>
        {/*<Row style={{marginTop: scale(5), paddingTop: scale(5), borderTopWidth: 1, borderColor: '#afafaf'}}>*/}
        {/*<View style={{borderBottomWidth: 1, borderColor: '#afafaf'}}>*/}
        {/*<View style={[styles.receiptTotalsRow, {borderBottomWidth: 1, borderColor: '#afafaf'}]}>*/}
        {/*<Text*/}
        {/*style={[styles.text, {fontSize: scaleText(8), color: '#afafaf', width: '50%'}]}>At betale</Text>*/}
        {/*<Text*/}
        {/*style={[styles.text, {fontSize: scaleText(8), color: '#afafaf', width: '50%', textAlign: 'right'}]}>{props.total}</Text>*/}
        {/*</View>*/}
        {/*</View>*/}
        {/*</Row>*/}
        <Row style={{marginTop: scale(5), paddingTop: scale(5), borderTopWidth: 1, borderColor: '#afafaf'}}/>
        {Object.keys(props.discount).length > 0 && props.discount.value !== '-0' && props.discount.value !== 0 &&
        <Row>
            <View style={styles.receiptTotalsRow}>
                <View style={[styles.receiptTotalsRow]}>
                    <View style={[{width: '70%'}]}>
                        <Text style={[styles.text, {
                            fontSize: scaleText(12), color: '#6a6a6a'}]}>{props.intl !== undefined && props.intl.messages["app.discount"].toUpperCase()} {props.discount.name}</Text>
                    </View>
                    <Text
                        style={[styles.text, {
                            fontSize: scaleText(12), color: '#6a6a6a', width: '30%', textAlign: 'right'
                        }]}>{toFixedLocale(props.discount.value)}</Text>
                </View>
            </View>
        </Row>
        }
        {props.total !== '' &&
        <Row>
            <View style={styles.receiptTotalsRow}>
                <View style={[styles.receiptTotalsRow]}>
                    <Text
                        style={[styles.text, {fontSize: scaleText(14), color: '#6a6a6a', width: '50%'}]}>
                        {props.intl !== undefined && props.intl.messages["app.paid"].toUpperCase()}</Text>
                    <Text
                        style={[styles.text, {
                            fontSize: scaleText(14),
                            color: '#6a6a6a',
                            width: '50%',
                            textAlign: 'right'
                        }]}>
                        {toFixedLocaleDa(props.total)}</Text>
                </View>
            </View>
        </Row>
        }
        {props.card !== undefined &&
        <Row>
            <View style={styles.receiptTotalsRow}>
                <View style={[styles.receiptTotalsRow]}>
                    <Text
                        style={[styles.text, {
                            fontSize: scaleText(9),
                            color: '#6a6a6a',
                            width: '100%'
                        }]}>{props.card}</Text>
                </View>
            </View>
        </Row>
        }
    </View>
);

export function toLocale(num) {
    return num.toLocaleString('da-DK', {minimumFractionDigits: 2, maximumFractionDigits: 2})
}

export function toFixedLocale(num) {
    return parseFloat(num).toFixed(2).toString().replace('.', ',')
}

export function toFixedLocaleDa(num) {
    let parts = parseFloat(num).toFixed(2).toString().split('.');
    parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, '.');
    return parts.join(',');
}

export function guid() {
    return guid4() + '-' + guid4() + '-' + guid4() + '-' + guid4();
}

export function guid4() {
    return Math.floor((1 + Math.random()) * 0x10000).toString(16).substring(1);
}

export const ReceiptItem = (props) => (props.data !== undefined &&

    <Col onPress={props.onPress}>
        <Row size={1} style={styles.closedReceiptHeader}>
            <Col>
                <Text style={{color: 'white', fontSize: scaleText(10)}}>
                    {moment.unix(props.data.timestamp).format('DD.MM.YYYY')}
                </Text>
            </Col>
            <Col>
                <Text style={{
                    textAlign: 'right',
                    color: 'white',
                    fontSize: scaleText(10)
                }}>KL: {moment.unix(props.data.timestamp).format('HH:mm:ss')}</Text>
            </Col>
        </Row>
        <Row size={3} style={styles.closedReceiptBody}>
            <Col>
                {props.data.table !== undefined && props.data.table.length > 0 &&
                <Row size={3}>
                    <Text style={{color: 'white', fontSize: scaleText(16), fontFamily: 'Gotham-Bold'}}>{props.data.number}</Text>
                </Row>}
                <Row size={4}>
                    <Text style={{color: 'white', fontSize: scaleText(22), fontFamily: 'Gotham-Bold'}}>{props.data.table}</Text>
                </Row>
                {props.data.customer_name !== undefined && props.data.customer_name.length > 0 &&
                <Row size={2.5}>
                    <Text style={{
                        color: 'white',
                        fontSize: scaleText(20),
                        fontFamily: 'Gotham-Bold'
                    }}>{props.data.customer_name ? props.data.customer_name.toUpperCase() : ''}</Text>
                </Row>
                }
            </Col>
        </Row>
        <Row size={3}>
            <ReceiptTotals total={props.data.total} discount={props.data.discount} small={true}
                           style={{marginRight: 15, marginBottom: scale(5)}} intl={props.intl}/>
        </Row>
    </Col>
);