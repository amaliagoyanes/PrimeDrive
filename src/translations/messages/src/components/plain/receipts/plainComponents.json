[
  {
    "id": "app.totalUpper",
    "defaultMessage": "TOTAL"
  },
  {
    "id": "app.paid",
    "defaultMessage": "Paid"
  }
]