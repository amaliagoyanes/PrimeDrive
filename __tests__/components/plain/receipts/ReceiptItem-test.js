'use strict';

import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import {ReceiptItem} from '../../../../src/components/plain/receipts/plainComponents';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

let data = {
    timestamp: 1483228800,
    number: 123,
    total: 1200,
    discount: 20
};

it('renders correctly', () => {
    const tree = shallow(
        <ReceiptItem data={data} onPress={jest.fn}/>);
    expect(tree).toMatchSnapshot();
    expect(tree.find('ReceiptTotals').length).toEqual(1)
});

let data2 = {
    timestamp: 1483228800,
    number: 123,
    total: 1200,
    discount: 20,
    customer_name: 'Customer'
};
it('renders correctly with customer name', () => {
    const tree = shallow(
        <ReceiptItem data={data2} onPress={jest.fn}/>);
    expect(tree).toMatchSnapshot();
    expect(tree.find('ReceiptTotals').length).toEqual(1)
});
