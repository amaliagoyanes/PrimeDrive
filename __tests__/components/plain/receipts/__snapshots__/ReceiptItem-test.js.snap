// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ColumnNB
  onPress={[Function]}
>
  <RowNB
    size={1}
    style={
      Object {
        "marginLeft": 15,
        "marginRight": 15,
        "marginTop": 15,
      }
    }
  >
    <ColumnNB>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 8,
          }
        }
      >
        01.01.2017
      </Text>
    </ColumnNB>
    <ColumnNB>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 8,
            "textAlign": "right",
          }
        }
      >
        KL: 
        03:00:00
      </Text>
    </ColumnNB>
  </RowNB>
  <RowNB
    size={3}
    style={
      Object {
        "marginLeft": 15,
      }
    }
  >
    <ColumnNB>
      <RowNB
        size={4}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 11,
            }
          }
        >
          123
        </Text>
      </RowNB>
      <RowNB
        size={6}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontFamily": "Gotham-Bold",
              "fontSize": 12,
            }
          }
        />
      </RowNB>
    </ColumnNB>
  </RowNB>
  <RowNB
    size={4}
  >
    <ReceiptTotals
      discount={20}
      small={true}
      style={
        Object {
          "marginBottom": 5,
          "marginRight": 15,
        }
      }
      total={1200}
    />
  </RowNB>
</ColumnNB>
`;

exports[`renders correctly with customer name 1`] = `
<ColumnNB
  onPress={[Function]}
>
  <RowNB
    size={1}
    style={
      Object {
        "marginLeft": 15,
        "marginRight": 15,
        "marginTop": 15,
      }
    }
  >
    <ColumnNB>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 8,
          }
        }
      >
        01.01.2017
      </Text>
    </ColumnNB>
    <ColumnNB>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 8,
            "textAlign": "right",
          }
        }
      >
        KL: 
        03:00:00
      </Text>
    </ColumnNB>
  </RowNB>
  <RowNB
    size={3}
    style={
      Object {
        "marginLeft": 15,
      }
    }
  >
    <ColumnNB>
      <RowNB
        size={4}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 11,
            }
          }
        >
          123
        </Text>
      </RowNB>
      <RowNB
        size={6}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontFamily": "Gotham-Bold",
              "fontSize": 12,
            }
          }
        >
          CUSTOMER
        </Text>
      </RowNB>
    </ColumnNB>
  </RowNB>
  <RowNB
    size={4}
  >
    <ReceiptTotals
      discount={20}
      small={true}
      style={
        Object {
          "marginBottom": 5,
          "marginRight": 15,
        }
      }
      total={1200}
    />
  </RowNB>
</ColumnNB>
`;
